### ISSUE TITLE
Initializer default values make it very easy to accidentally create 2 distinct ApolloStore's

### ISSUE THREAD

**Issue by namolnad** [apollographql/apollo-ios#1460]
## Bug report

Currently it's very easy, through default initializer values, to end up with multiple ApolloStore's and the inability to properly invalidate your cache. I just spent a fair amount of time debugging this issue, which I think could be resolved by removing some default values in one or both the `ApolloClient`'s and `LegacyInterceptorProvider`'s initializers. The problem is as follows:

``` swift
let interceptorProvider: LegacyInterceptorProvider = .init() // More likely to use your own subclass for authentication
let transport: RequestChainNetworkTransport = .init(
    interceptorProvider: interceptorProvider,
    endpointURL: baseUrl.appendingPathComponent("graphql")
)
let client: ApolloClient = .init(
    networkTransport: transport
)
```

The above code creates two separate `ApolloStore`'s with default caches of `InMemoryNormalizedCache()` for both the `LegacyInterceptorProvider` and for the `ApolloClient`. When going to invalidate your cache here, you will invalidate one of the caches (the ApolloStore one), but the InterceptorProvider's cache will remain untouched, which is where results seem to be fetched from, thus making it impossible to actually invalidate the cache. Without these default values, I believe this problem would be obvious to the developer.

## Versions

Please fill in the versions you're currently using: 

- `apollo-ios` SDK version: 0.34.0
- Xcode version: 12.0.1
- Swift version: 5.3

## Steps to reproduce

Initialize the ApolloClient using default values, as above.

**Comment by designatednerd** [comment]
Yeah this is basically a dupe of #1438 - I'm going to take a look at your solution, I was still pondering exactly how to get this set up in a friendlier way.

**Comment by namolnad** [comment]
Yup, definitely a dupe — sorry for missing that and thanks for taking a look!

**Comment by designatednerd** [comment]
This has shipped with `0.36.0`!
